assignment/tool/src/webapp/js/studentViewSubmission.js

```diff
@@ -31,3 +35,179 @@ ASN_SVS.undoCancel = function()
+{
+    var endpoint = "/direct/assignment/removeTimeSheet.json";
+    var params = {};
+    var ts = document.getElementsByName("selectedTimesheet");
+    var tsassignmentId = document.getElementById("assignmentId").value;
+
+    params.selectedTimeSheets = Array.apply(null, ts).filter((el) => el.checked).map((el) => el.value);
+    params.assignmentId = tsassignmentId;
+
+    button.classList.add("spinButton");
+    button.disabled = true;
+
+    ASN_TS_API._POST(endpoint, params, onSuccess, onError);
+};
+
+//SAK-43155
+ASN_TS_API._GET = function(url, data, onSuccess, onError, onComplete)
+{
+    $.ajax(
+    {
+        type: "GET",
+        url: url,
+        data: data,
+        cache: false,
+        success: onSuccess || $.noop,
+        error: onError || $.noop,
+        complete: onComplete || $.noop
+    });
+};
+
+//SAK-43155
+ASN_TS_API._POST = function(url, data, onSuccess, onError, onComplete)
+{
+    $.ajax(
+    {
+        type: "POST",
+        url: url,
+        data: data,
+        success: onSuccess || $.noop,
+        error: onError || $.noop,
+        complete: onComplete || $.noop
+    });
+};
+
+//SAK-43155
+ASN.switchTimesheetTab = function( source )
+{
+    if (['tabAssignment', 'tabTimeSheet'].includes(source)) {
+        document.getElementById('tabAssignmentCurrent').classList.toggle('hidden');
+        document.getElementById('tabAssignmentRef').classList.toggle('hidden');
+        document.getElementById('tabTimeSheetRef').classList.toggle('hidden');
+        document.getElementById('tabTimeSheetCurrent').classList.toggle('hidden');
+        document.getElementById('StudentAssignmentCurrent').classList.toggle('hidden');
+        document.getElementById('StudentTimesheetCurrent').classList.toggle('hidden');
+    }
+};
+
+//SAK-43155
+ASN.handleAjaxAddSuccess = function(data)
+{
+    if (data.error && data.error.message) {
+        var button = document.getElementById("btnTimesheetAdd");
+        button.classList.remove("spinButton");
+        button.disabled = false;
+        var alertTsheetAddRecord = document.getElementById("alertTsheetAddRecord");
+        alertTsheetAddRecord.classList.toggle('hidden');
+        alertTsheetAddRecord.innerHTML= window.i18nProgresa[data.error.message];		
+    } else {
+        ASN.submitForm( 'addSubmissionForm', 'view', null, null );
+    }
+};
+
+//SAK-43155
+ASN.handleAjaxRemoveSuccess = function(data)
+{
+    if (data.error && data.error.message) {
+        var button = document.getElementById("btnTimesheetDelete");
+        button.classList.remove("spinButton");
+        button.disabled = false;
+        var alertTsheetDelRecord = document.getElementById("alertTsheetDelRecord");
+        alertTsheetDelRecord.classList.toggle('hidden');
+        alertTsheetDelRecord.innerHTML= window.i18nProgresa[data.error.message];
+    } else {
+        ASN.submitForm( 'addSubmissionForm', 'view', null, null );
+    }
+};
+
+//SAK-43155
+ASN.addHandleAjaxError = function(xhr)
```

Comment:
this function needs to be named better, so we know it's about timesheets.

Reply:
`ASN.handleAjaxAddSuccess    ->  ASN.tsHandleAjaxAddSuccess
ASN.handleAjaxRemoveSuccess ->  ASN.tsHandleAjaxRemoveSuccess
ASN.addHandleAjaxError      ->  ASN.tsAddHandleAjaxError
ASN.removeHandleAjaxError   ->  ASN.tsRemoveHandleAjaxError`